src/entities/identity.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Mode for identifying users in the Purchases SDK.
+ * @public
+ */
+export enum IdentityMode {
```

Comment:
Unsure about this name and the usage of the "identity" word.

Reply:
I'm torn on it frankly. Part of me thinks "we do talk about identity as a concept [in our docs"](https://www.revenuecat.com/docs/customers/user-ids), but also, it's still an awkward and not intuitive thing to write when using the API. 

I wonder if things might look better if we instead take inspiration from the observer mode changes? 

We used to ask people to enable or disable "observer mode", and that introduced a lot of questions and awkwardness, so we updated the API to be much more explicit about what it means in practice: 

https://www.revenuecat.com/docs/migrating-to-revenuecat/sdk-or-not/finishing-transactions

So now it looks like this (note the last line): 

```swift
Purchases.configure(
    with: .init(withAPIKey: "<public_sdk_key>")
          .with(purchasesAreCompletedBy: .myApp, storeKitVersion: .storeKit2)
)
```

Maybe we could do something similar, like `appUserIDsAreProvidedBy: myApp | revenueCat`? WDYT?

Reply:
I guess it looks different / more awkward in that you currently pass in the appUserID itself... maybe if we called it `RevenueCatAutoGenerated` or something? Let's bring this up in `#guild-sdks` for some team-wide wisdom

Reply:
I've given it a shot, adding an additional parameter `appUserIDsAreProvidedBy` and using overloads to avoid breaking changes to the API so we can still call configure without the new parameter, and avoiding passing an app user ID when `appUserIDsAreProvidedBy == AppUserIDsProvider.RevenueCat`. 
I've not an expert in typescript but I think it works as expected. 
Let me know what you think!

Reply:
Gave it another shot and now we have named params